### PR TITLE
[Chore] Use phoenix releases for deployment

### LINKED ISSUE #47: [Chore] Use phoenix releases for deployment
Phoenix and Elixir allow packaging apps as a so [called release](https://hexdocs.pm/phoenix/releases.html). This release contains everything required to run the application. 

We'll also switch to Debian bullseye to stay in line with what ```mix phx.gen.release --docker``` proposes.

### PR DESCRIPTION
**Pull request by nwittstruck**
This PR switches to phoenix generated releases. See https://hexdocs.pm/phoenix/releases.html

This PR therefore switches from Alpine to Debian Bullseye.

Closes #47

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,91 +1,135 @@
...
-    ;
+RUN apt-get update -y && apt-get install -y libstdc++6 openssl libncurses5 locales \
+  && apt-get clean && rm -f /var/lib/apt/lists/*_*
 
-WORKDIR $APP_PATH
+# Set the locale
+RUN sed -i '/en_US.UTF-8/s/^# //g' /etc/locale.gen && locale-gen
 
-RUN chown nobody:nobody $APP_PATH/
+ENV LANG en_US.UTF-8
```
Comment (JannikStreek) [inline Dockerfile:120]: is all this US stuff intentional?

File: Dockerfile [hunk 2]
```diff
@@ -1,91 +1,135 @@
...
-      ncurses-libs \
-      openssl \
-      postgresql-client \
-    ;
+RUN apt-get update -y && apt-get install -y libstdc++6 openssl libncurses5 locales \
+  && apt-get clean && rm -f /var/lib/apt/lists/*_*
 
-WORKDIR $APP_PATH
+# Set the locale
+RUN sed -i '/en_US.UTF-8/s/^# //g' /etc/locale.gen && locale-gen
```
Comment (JannikStreek) [inline Dockerfile:118]: same here

File: Dockerfile [hunk 3]
```diff
@@ -1,91 +1,135 @@
...
-ENV HOME=$APP_PATH
+# Only copy the final release from the build stage
+COPY --from=builder --chown=nobody:root /app/_build/${MIX_ENV}/rel/qrstorage ./
 
-ENTRYPOINT ["sh", ".docker/entrypoint.release.sh"]
+USER nobody
 
-EXPOSE 8000
\ No newline at end of file
+CMD ["/app/bin/server"]
```
Comment (gerardo-navarro) [inline Dockerfile:135]: Attention: This startup script does not execute migrations as it was done before. The previous start up script `.docker/entrypoint.release.sh` executed also the migrations.

**Review by gerardo-navarro — COMMENTED** [review]
Maybe you can also extend the `README.md` documentation regarding the production steps.

**Pull request merged**